src/index.ts

```diff
@@ -55,12 +120,18 @@ enum InteractionResponseFlags {
  * @returns Whether or not validation was successful
  */
 async function verifyKey(
-  rawBody: Buffer,
-  signature: string,
-  timestamp: string,
-  clientPublicKey: string,
+  body: Uint8Array | ArrayBuffer | Buffer | string,
+  signature: Uint8Array | ArrayBuffer | Buffer | string,
+  timestamp: Uint8Array | ArrayBuffer | Buffer | string,
+  clientPublicKey: Uint8Array | ArrayBuffer | Buffer | string,
 ): Promise<boolean> {
-  return await edVerify(signature, Buffer.concat([Buffer.from(timestamp, 'utf-8'), rawBody]), clientPublicKey);
+  const timestampData = valueToUint8Array(timestamp);
+  const bodyData = valueToUint8Array(body);
+  const message = concatUint8Arrays(timestampData, bodyData);
+
+  const signatureData = valueToUint8Array(signature, 'hex');
+  const publicKeyData = valueToUint8Array(clientPublicKey, 'hex');
+  return await nacl.sign.detached.verify(message, signatureData, publicKeyData);
```

Comment:
this method is synchronous, and calling this with await implicitly handles the errors it throws. should we make this method synchronous and handle the errors thrown explicitly?